mypy/newsemanal/semanal_pass1.py

```diff
@@ -40,16 +44,35 @@ def visit_file(self, file: MypyFile, fnam: str, mod_id: str, options: Options) -
         self.cur_mod_id = mod_id
         self.cur_mod_node = file
         self.options = options
+        self.is_global_scope = True
 
         for i, defn in enumerate(file.defs):
+            if isinstance(defn, (ClassDef, FuncDef)):
+                self.is_global_scope = False
             defn.accept(self)
+            self.is_global_scope = True
             if isinstance(defn, AssertStmt) and assert_will_always_fail(defn, options):
                 # We've encountered an assert that's always false,
                 # e.g. assert sys.platform == 'lol'.  Truncate the
                 # list of statements.  This mutates file.defs too.
                 del file.defs[i + 1:]
                 break
 
+    def visit_import_from(self, node: ImportFrom) -> None:
+        if self.is_global_scope:
+            node.is_top_level = True
```

Comment:
What about just saying `node.is_top_level = self.is_global_scope` (here and below)?